specifications/drtm-handler-and-table.rst

```diff
@@ -0,0 +1,310 @@
+
+.. code-block::
+   :linenos:
+
+    struct drtm_table {
+        uint32_t magic; /* 0x4452544d */
+        uint16_t revision;
+        uint16_t architecture;
+        uint32_t table_size;
+        uint32_t table_max_size
+        struct drtm_entry_hdr entries[0];
+    };
+
+DRTM Table Entry
+""""""""""""""""
+
+.. code-block::
+   :linenos:
+
+    struct drtm_entry_hdr {
+        uint16_t tag;
+        uint16_t size;
+    };
+
+
+DRTM Dynamic Launch Configuration
+"""""""""""""""""""""""""""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x0008 */
+    struct drtm_entry_dl_info {
+        struct drtm_entry_hdr hdr;
+        uint64_t dl_handler;
+        uint64_t dce_base;
+        uint32_t dce_size;
+        uint64_t dlme_entry;
+    };
+
+DRTM Log
+""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x0009 */
+    struct drtm_entry_log {
+        struct drtm_entry_hdr hdr;
+        uint16_t format;
+        uint16_t reserved;
+        uint64_t addr;
+        uint32_t size;
+    };
+
+
+DRTM Policy
+"""""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x000a */
+    struct drtm_entry_policy {
+        struct drtm_entry_hdr hdr;
+        uint16_t revision;
+        uint16_t reserved;
+        uint16_t algorithms
+        uint16_t nr_entries;
+        struct policy_entry entries[0];
+    };
+  
+
+DRTM Policy Entry
+"""""""""""""""""
+
+A policy entry represents an entity that the SL Entry is being requested to
+measure. As an SL Entry is able to measure an attribute of a kernel, that
+attribute will be published as an entity type. A general "memory location"
+entity type is available for measuring a range of memory.
+
+.. code-block::
+   :linenos:
+
+    struct drtm_policy_entry {
+        uint16_t pcr;
+        uint16_t entity_type;
+        uint64_t entity; /* entity addr if it cannot be infered from the type */
+        uint16_t size;
+        uint16_t pad;
+        char evt_info[8];
+    };
+
+Intel TXT Info
+""""""""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x000b */
+    struct drtm_entry_intel_info {rp-openxt-amd-grub-debug-serial
+        struct drtm_entry_hdr hdr;
+        uint16_t revision;
+        uint16_t reserved;
+        uint32_t boot_params_addr;
+        uint64_t saved_misc_enable_msr;
+        struct {
+            u64 default_mem_type;
+	          u64 mtrr_vcnt;
+            struct {
+              u64 mtrr_physbase;
+              u64 mtrr_physmask;
+            } mtrr_pair[32];
+        } saved_bsp_mtrrs;
+        uint32_t ap_wake_block;
+        uint32_t ap_wake_block_size;
+    };
+
+AMD SKINIT Info
+"""""""""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x000c */
+    struct drtm_entry_amd_info {
+        struct drtm_entry_hdr hdr;
+    };
+
+ARM Info
+""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x000d */
+    struct drtm_entry_arm_info {
+        struct drtm_entry_hdr hdr;
+    };
+
+EFI Info
+""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x000e */
```

Comment:
We will reserve this EFI table tag but we will not define the table in the document for now. Add a note about why we are doing this to the doc.